repl/src/main/scala/ammonite/repl/Main.scala

```diff
@@ -71,21 +81,52 @@ case class Main(predef: String = "",
     res
   }
 
+
   /**
     * Run a Scala script file! takes the path to the file as well as an array
     * of `args` and a map of keyword `kwargs` to pass to that file.
     */
   def runScript(path: Path,
                 args: Seq[String],
-                kwargs: Map[String, String]): Res[Imports] = {
-
-    val repl = instantiateRepl()
+                kwargs: Map[String, String],
+                storage: Storage): Res[Seq[ImportData]] = {
+    val outBuffer = mutable.Buffer.empty[String]
+    val warningBuffer = mutable.Buffer.empty[String]
+    val errorBuffer = mutable.Buffer.empty[String]
+    val infoBuffer = mutable.Buffer.empty[String]
+    val prompt = Ref("@ ")
+    val colors = Ref[Colors](Colors.Default)
+    val frontEnd = Ref[FrontEnd](AmmoniteFrontEnd(Filter.empty))
+    val printer = Printer(
+      outBuffer.append(_),
+      warningBuffer.append(_),
+      errorBuffer.append(_),
+      infoBuffer.append(_)
+    )
+    val predef = Main.maybeDefaultPredef(defaultPredef, Main.defaultPredefString)
+
+    val interp: Interpreter = new Interpreter(
+      prompt,
+      frontEnd,
+      frontEnd().width,
+      frontEnd().height,
+      colors,
+      printer,
+      storage,
+      new History(Vector()),
+      predef,
+      wd,
+      Nil
+    )
     val (pkg, wrapper) = Util.pathToPackageWrapper(path, wd)
-    repl.interp.processModule(ImportHook.Source.File(path), read(path), wrapper, pkg) match{
+    interp.processModule(ImportHook.Source.File(path), read(path), wrapper, pkg) match {
       case x: Res.Failing => x
-      case Res.Success(imports) =>
-        repl.interp.init()
-        imports.value.find(_.toName == "main") match {
+      case Res.Success(processedData) =>
+        val Imports(imports) = processedData
```

Comment:
Rather than pattern match, extracting `imports`, and then wrapping it back in a `Res.Success`, you should just return `processedData` directly below